airbyte-integrations/bases/debezium/src/main/java/io/airbyte/integrations/debezium/internals/DebeziumRecordIterator.java

```diff
@@ -214,9 +210,6 @@ private void throwExceptionIfSnapshotNotFinished() {
    */
   @VisibleForTesting
   protected T getHeartbeatPosition(final ChangeEvent<String, String> heartbeatEvent) {
-    if (heartbeatEvent == null) {
-      return null;
-    }
```

Comment:
This is not redundant is an early exit to reduce unneeded computation and object creation.

Reply:
> This is not redundant is an early exit to reduce unneeded computation and object creation.

@sergio-ropero please take a look [here](https://github.com/airbytehq/airbyte/blob/c1a4a6f3d3b539a4bed435d82cc4253130df6957/airbyte-integrations/bases/debezium/src/main/java/io/airbyte/integrations/debezium/internals/DebeziumRecordIterator.java#L177)
this check for null is definitely not needed since it can never occur

Reply:
Yes and no, so it will be never null if you include it inside an if statement that previously checks it's not null.
I think we can remove it, but we need to ensure that we always include that if stamente.

Reply:
> Yes and no, so it will be never null if you include it inside an if statement that previously checks it's not null.
> I think we can remove it, but we need to ensure that we always include that if statement.

i don't understand what do you mean to  "ensure that we always include that if statement" if we will never get null here

Reply:
What I mean is that they are 2 different functions, and it will never be null because in our current code we are including the function `getHeartbeatPosition` inside the if statement that uses `isHeartbeatEvent`.

That means that the function `getHeartbeatPosition` can be called (in other scenarios) with a null object, except if again we include it inside a block code with the same if statement.